osu.Game/Skinning/RulesetSkinProvidingContainer.cs

```diff
@@ -64,7 +64,7 @@ protected virtual void UpdateSkins()
         protected ISkin GetRulesetTransformedSkin(ISkin skin)
         {
             if (!(skin is LegacySkin))
-                return null;
+                return skin;
```

Comment:
I'm not sure how you came up with this. Did you think about this one or just make the change and see tests turn green before pushing?

Reply:
I've done it this way to make it a bit cleaner from usage perspective, but I could rename this to `GetLegacyRulesetTransformedSkin` and not do this here.

Reply:
In its current state it's returning the skin itself in cases where it's not required to make a transformer. Also the `Legacy` conditional is being done *outside* this method in at least one case.